sdk/inc/azure/core/az_span.h

```diff
@@ -441,6 +441,46 @@ AZ_NODISCARD az_result az_span_u64toa(az_span destination, uint64_t source, az_s
 AZ_NODISCARD az_result
 az_span_dtoa(az_span destination, double source, int32_t fractional_digits, az_span* out_span);
 
+/******************************  NON-CONTIGUOUS SPAN  */
+
+/**
+ * @brief Defines a container of required and user-defined fields that provide the
+ * necessary information and parameters for the implementation of the #az_span_allocator_fn
+ * callback.
+ */
+typedef struct
+{
+  int32_t remaining_size; ///< The amount of space left within the previously provided destination,
+                          ///< which can be used to infer the number of bytes that have already been
+                          ///< written into that #az_span.
+  int32_t required_size; ///< The minimum length of the destination #az_span required to be provided
+                         ///< by the callback. If 0, any non-empty sized buffer must be returned.
+  void* user_context; ///< Any struct or set of fields that are provied by the user for their
```

Comment:
Yes, this is the user_context void* that should be passed to *_init

Reply:
Sounds good. I know you had reservations with exposing the other fields too eagerly, on the `az_allocator_context`, but for some of the callback function implementations, such information is necessary and only components like the json builder know/can provide that.